[HM33-Stu] iP

HM33-Stu opened the pull request:
# Hari-Bot

"Need to Hari-Up? Then, Hari-Up and use Hari-Bot"

— Hari (Founder, Creator)

Hari-Bot is a text-based task management tool designed to make it easy for users to manage their tasks efficiently. It is fast, super fast, and easy to learn. Here's how you can get started with Hari-Bot:

1. **Download**: Simply download Hari-Bot
2. **Installation**: Double-click the downloaded file to install Hari-Bot.
3. **Add Tasks**: Start adding your tasks to Hari-Bot.
4. **Let it Manage**: Sit back and let Hari-Bot manage your tasks for you 😉

HariPro is completely free to use and offers the following features:

- **Managing Tasks**: Easily add, edit, and delete tasks.
- **Managing Deadlines** (Coming Soon): Stay organized by managing deadlines efficiently.
- **Reminders** (Coming Soon): Get timely reminders to stay on top of your tasks.



```
public static void main(String[] args) {

    new Duke(LISTPATH).run();
}

HM33-Stu committed:
Level-0

HM33-Stu committed:
Delete src/main/java/Duke.java

HM33-Stu committed:
Level-1

HM33-Stu committed:
Merge branch 'master' of https://github.com/HM33-Stu/ip

HM33-Stu committed:
Level-2

HM33-Stu committed:
Level-3

HM33-Stu committed:
A-Classes

HM33-Stu committed:
Level-4

HM33-Stu committed:
A-Inheritance

HM33-Stu committed:
A-TextUiTesting

HM33-Stu committed:
Level-5

HM33-Stu committed:
A-Exceptions

HM33-Stu committed:
Level-6

HM33-Stu committed:
Added more error handling

HM33-Stu committed:
Added more error handling

HM33-Stu committed:
Level-7

HM33-Stu committed:
Merge branch 'branch-Level-7'

MaYuehan commented on src/main/java/Hari.java:
```diff
@@ -1,4 +1,9 @@
+import java.io.IOException;
 
 // Class for the chatbot itself
 class handlerbot {
```
Perhaps renaming handlerbot to HandlerBot to follow PascalCase convention?

HM33-Stu replied:
Thanks! Have made the necessary amendments for this and the others you highlighted

MaYuehan commented on src/main/java/Hari.java:
```diff
@@ -93,14 +98,130 @@ public boolean summaryempty() {
+    }
+
     // Function that handles the greeting message
     public void messagegreeting() {
```
renaming the messagegreeting to messageGreeting to follow camelCase convention?

MaYuehan commented on src/main/java/Hari.java:
```diff
@@ -93,14 +98,130 @@ public boolean summaryempty() {
 
+
     // Function that handles the exit message
     public void messagefarewell() {
```
renaming the messagefarewell to messageFarewell to follow camelCase convention?

MaYuehan commented on src/main/java/Hari.java:
```diff
@@ -93,14 +98,130 @@ public boolean summaryempty() {
+    }
+
+    // Custom exception class for handling corrupted data
+    class CorruptedDataException extends Exception {
```
Consider moving CorruptedDataException to its own separate class file?

MaYuehan commented on src/main/java/Hari.java:
```diff
@@ -93,14 +98,130 @@ public boolean summaryempty() {
+                writer.write(task.taskstatus() + " | " + (task.completion ? "1" : "0") + " | " +
+                        task.summarystatus() + " | " +
+                        (task.taskstatus().equals("D") ? task.deadlinestatus() : "") + " | " +
+                        (task.taskstatus().equals("E") ? task.timerstartstatus() + " to " + task.timerendstatus() : "") + "\n");
```
prehaps divide this long line into two lines?

HM33-Stu committed:
branch-Level-8

HM33-Stu committed:
Level-8

HM33-Stu committed:
Merge branch 'branch-Level-8'

HM33-Stu committed:
A-MoreOOP

HM33-Stu committed:
A-Packages

HM33-Stu committed:
A-Gradle

HM33-Stu committed:
A-JUnit

HM33-Stu committed:
A-Jar

HM33-Stu committed:
A-JavaDoc

HM33-Stu committed:
A-CodingStandard

HM33-Stu committed:
Level-9

HM33-Stu committed:
Merge branch 'branch-A-JavaDoc'

HM33-Stu committed:
Merge branch 'branch-A-CodingStandard'

HM33-Stu committed:
Merge branch 'branch-Level-9'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // To mark as completed
+        public void completionMark() {
```
I think the naming of methods should be verbs instead of nouns. So maybe something like 'markAsComplete' would be better.

HM33-Stu replied:
Thanks! Have amended my method names (as best as possible) to reflect the purpose of the function itself.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // To unmark completed tasks
+        public void incompletionMark() {
```
Maybe 'markAsIncomplete'.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // For displaying the X or [ ] depending on completion status
+        public String completionStatus() {
```
Maybe 'getStatus'.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // For displaying task description
+        public String summaryStatus() {
```
Maybe 'getDescription'.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // For displaying the task type
+        public String taskStatus() {
```
Maybe 'getTaskType'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // For displaying start time of event tasks
+        public String timerStartStatus() {
```
Maybe 'getStartTime'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // For displaying end time of event tasks
+        public String timerEndStatus() {
```
Maybe 'getEndTime'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // For displaying deadline of deadline tasks
+        public String deadlineStatus() {
```
Maybe 'getDeadline'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+        }
+
+        // Additional method to check if task description is empty
+        public boolean summaryEmpty() {
```
For booleans, I think it should be named to sound like booleans. So in this case, maybe 'isDescriptionEmpty' could be one way to name this.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+    }
+
+    // Function to delete a task
+    public void taskDeleter(int taskNumber) {
```
Maybe 'deleteTask'.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+    }
+
+    // Function to unmark previously marked as completed task
+    public void incompletionMark(int taskRecorder) {
```
Maybe 'markAsIncomplete'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+    }
+
+    // Function to mark task as completed
+    public void completionMark(int taskRecorder) {
```
Maybe 'markAsComplete'

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+
+    // Function to add tasks
+    // No modification done to userEchoedInput function as not all inputs are tasks
+    public void additionTaskings(String taskings) {
```
Maybe 'addTask'.

Derekljh commented on src/main/java/Hari.java:
```diff
@@ -0,0 +1,535 @@
+
+    // Function to display tasks
+    // No modification done to userEchoedInput function as not all inputs are tasks
+    public void taskingsDisplay() {
```
Maybe 'listTasks'

Derekljh reviewed (commented):
In my opinion, naming of the methods definitely needs some work. However, since it is still early in the individual project, it can still be remedied easily. Best of luck for the rest of your tasks!

HM33-Stu committed:
Addressed peer PR review comments

HM33-Stu committed:
branch-Level-10

HM33-Stu committed:
Delete src/main/java/Hari.java

HM33-Stu committed:
Fixed application loading errors

HM33-Stu committed:
Merge branch 'master' of https://github.com/HM33-Stu/ip

HM33-Stu committed:
Completed some tests

HM33-Stu committed:
Updated

HM33-Stu committed:
Updated

HM33-Stu committed:
Update user interface code

This commit updates the user interface elements to improve clarity and
usability. By adjusting the layout and styling, the interface is made
more intuitive for users.

These changes enhance the overall user
experience and make the application more visually appealing.

HM33-Stu committed:
BCD-Extension

HM33-Stu committed:
Update code quality

This commit updates some minor elements in the code to improve clarity and
usability. By adjusting the layout and styling, the code is made more readadble.

These changes enhance the overall project and resuability.

HM33-Stu committed:
Push A-Assertions branch to fork

HM33-Stu committed:
Push A-CodeQuality branch to fork

HM33-Stu committed:
Merge pull request #2 from HM33-Stu/A-CodeQuality

A-CodeQuality

HM33-Stu committed:
Merge pull request #3 from HM33-Stu/A-Assertions

A-Assertions

HM33-Stu committed:
Addition of Gradle files

HM33-Stu committed:
Merge pull request #4 from HM33-Stu/A-Assertions

Addition of Gradle files

HM33-Stu committed:
A-UserGuide

HM33-Stu committed:
Merge branch 'master' of https://github.com/HM33-Stu/ip

HM33-Stu committed:
Amendments to User Guide

HM33-Stu committed:
Resolved checkstyle issues

HM33-Stu committed:
Fixed issues with UI

HM33-Stu committed:
Fixed user guide and more updates to Ui

HM33-Stu committed:
Renamed png to PNG

HM33-Stu committed:
Rename Ui.PNG to Ui.png

HM33-Stu committed:
Resolved .png issues

HM33-Stu committed:
Merge branch 'master' of https://github.com/HM33-Stu/ip

HM33-Stu committed:
Added in functionality where application exits upon bye command

HM33-Stu committed:
Image

HM33-Stu committed:
Resolved .png issues for Ui image

HM33-Stu committed:
Amended Ui image

HM33-Stu committed:
Cleaned up User Guide

HM33-Stu committed:
Update README.md

HM33-Stu committed:
Cleaned Up Javadoc Comments

HM33-Stu committed:
A-BetterGui

HM33-Stu committed:
A-Personality

HM33-Stu committed:
Added in delay for application exit

HM33-Stu committed:
Amended JavaDocs and formatting

HM33-Stu committed:
Resolved checkstyle issues

HM33-Stu committed:
Resolved JavaDocs errors

HM33-Stu committed:
Resolved JavaDocs and Checkstyle Issues

HM33-Stu committed:
Addressed CheckStyle issues

HM33-Stu committed:
Added back in Text-Ui-Test files

HM33-Stu committed:
A-MoreTesting

HM33-Stu committed:
Fixed UI scrolling issues

HM33-Stu committed:
Cited References and Cleaned Up Checkstyle Issues

HM33-Stu committed:
Resolved Scroll Pane UI issues

HM33-Stu committed:
Update UI image

HM33-Stu committed:
Update README.md

HM33-Stu committed:
Update README.md
